library/core/src/alloc/layout.rs

```diff
@@ -366,17 +362,26 @@ impl Layout {
     /// // But you can manually make layouts which don't meet that rule.
     /// let padding_needed = Layout::from_size_align(6, 4).unwrap();
     /// let repeated = padding_needed.repeat(3).unwrap();
-    /// assert_eq!(repeated, (Layout::from_size_align(24, 4).unwrap(), 8));
+    /// assert_eq!(repeated, (Layout::from_size_align(22, 4).unwrap(), 8));
     /// ```
```

Comment:
While you're here and this is changing, can you please add a test for `.repeat(0)` too?  I think it's both useful to show the edge case for the documentation reader and because there's a new branch for it in the implementation.

Maybe something like
```suggestion
    /// 
    /// // Repeating an element zero times has zero size, but keeps the alignment (like `[T; 0]`)
    /// let repeated = normal.repeat(0);
    /// assert_eq!(repeated, (Layout::from_size_align(0, 4).unwrap(), 12));
    /// let repeated = padding_needed.repeat(0);
    /// assert_eq!(repeated, (Layout::from_size_align(0, 4).unwrap(), 8));
    /// ```
```